client/a8c-for-agencies/components/add-new-site-button/import-from-wpcom-modal/wpcom-sites-table.tsx

```diff
@@ -0,0 +1,190 @@
+import { WordPressLogo, JetpackLogo, Gridicon } from '@automattic/components';
+import { useDesktopBreakpoint } from '@automattic/viewport-react';
+import { CheckboxControl } from '@wordpress/components';
+import { Icon } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useState, useMemo, useCallback } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import TextPlaceholder from 'calypso/a8c-for-agencies/components/text-placeholder';
+import useFetchDashboardSites from 'calypso/data/agency-dashboard/use-fetch-dashboard-sites';
+import { urlToSlug } from 'calypso/lib/url/http-utils';
+import { useSelector } from 'calypso/state';
+import { getActiveAgencyId } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import getSites from 'calypso/state/selectors/get-sites';
+import getIsSiteWPCOM from 'calypso/state/selectors/is-site-wpcom';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import type { Site } from 'calypso/a8c-for-agencies/sections/sites/types';
+
+type SiteItem = {
+	id: number;
+	site: string;
+	date: string;
+	type: string;
+};
+
+const TypeIcon = ( { siteId }: { siteId: number } ) => {
+	const isWPCOM = useSelector( ( state ) => getIsSiteWPCOM( state, siteId ) );
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+
+	if ( isWPCOM ) {
+		return <Icon className="wpcom-sites-table__icon" icon={ <WordPressLogo /> } />;
+	}
+	if ( isJetpack ) {
+		return <Icon className="wpcom-sites-table__icon" icon={ <JetpackLogo /> } />;
+	}
+	return <Gridicon icon="minus" />;
+};
+
+export default function WPCOMSitesTable( {
+	selectedSites,
+	setSelectedSites,
+}: {
+	selectedSites: number[];
+	setSelectedSites: ( sites: number[] ) => void;
+} ) {
+	const translate = useTranslate();
+	const agencyId = useSelector( getActiveAgencyId );
+
+	const { data, isFetching } = useFetchDashboardSites( {
+		isPartnerOAuthTokenLoaded: false,
+		searchQuery: '',
+		currentPage: 1,
+		sort: {
+			field: '',
+			direction: '',
+		},
+		perPage: 100,
+		agencyId,
+		filter: {
+			issueTypes: [],
+			showOnlyFavorites: false,
+		},
+	} );
+
+	const sites = useSelector( getSites );
+
+	const isDesktop = useDesktopBreakpoint();
+
+	const [ dataViewsState, setDataViewsState ] = useState( initialDataViewsState );
+
+	// FIXME: This is a temporary solution to filter out sites that are already connected
+	const items = useMemo( () => {
+		return sites
+			.filter(
+				( site ) =>
+					site?.visible &&
+					! site.is_private &&
+					data?.sites.every( ( s: Site ) => s.blog_id !== site.ID )
```

Comment:
This will probably perform poorly on 1000+ sites. Maybe we can build a map of all agency site IDs (once) and filter it like the one below. Wdyt?

```
// Site ID map of all known managed sites.
const managedSites = {
     106: true
};

site.filter((site) => site?.visible && !site.is_private &&  managedSites[ site.ID ]);
```

Reply:
We will address this as a part of https://github.com/Automattic/automattic-for-agencies-dev/issues/762